content/post/2022-01-03-mlr3-package-updates-q4-2021/index.Rmd

```diff
@@ -0,0 +1,85 @@
+
+* Learners which are capable of resuming/continuing (e.g.,
+  learner `(classif|regr|surv).xgboost` with hyperparameter `nrounds` updated)
+  can now optionally store a stack of trained learners to be used to hotstart
+  their training. Note that this feature is still somewhat experimental.
+  See `HotstartStack` and #719.
+* New measures to score similarity of selected feature sets:
+  `sim.jaccard` (Jaccard Index) and `sim.phi` (Phi coefficient) (#690).
+* `predict_newdata()` now also supports `DataBackend` as input.
+* New function `install_pkgs()` to install required packages. This generic works
+  for all objects with a `packages` field as well as `ResampleResult` and
+  `BenchmarkResult` (#728).
+* New learner `regr.debug` for debugging.
+* New `Task` method `$set_levels()` to control how data with factor columns
+  is returned, independent of the used `DataBackend`.
+* Measures now return `NA` if prerequisite are not met (#699).
+  This allows to conveniently score your experiments with  multiple measures
+  having different requirements.
+* Feature names may no longer contain the special character `%`.
+
+
+# mlr3db 0.4.2 - https://github.com/mlr-org/mlr3db
+
+**Description: Data Base Backend for 'mlr3'**
+
+- Compatibility fixes with new duckdb version.
+
+
+# mlr3learners 0.5.1- - https://github.com/mlr-org/mlr3learners
+
+**Description: Recommended learners for mlr3**
+
+- `eval_metric()` is now explicitly set for xgboost learners to silence a
+  deprecation warning.
+- Improved how the added hyperparameter `mtry.ratio` is converted to `mtry` to
+  simplify tuning.
+- Multiple updates to hyperparameter sets.
+
+
+# mlr3pipelines 0.4.0 - https://github.com/mlr-org/mlr3pipelines
+
+**Description: Preprocessing Operators and Pipelines for 'mlr3'**
+
+* New operator `%>>!%` that modifies Graphs in-place.
+* New methods `chain_graphs()`, `concat_graphs()`, `Graph$chain()` as alternatives for `%>>%` and `%>>!%`.
+* New methods `pos()` and `ppls()` which create lists of PipeOps/Graphs and can be seen as "plural" forms of `po()` and `ppl()`.
+* `po()` S3-method for PipeOp class that clones a PipeOp object and optionally modifies its attributes.
+* `Graph$add_pipeop()` now clones the PipeOp being added.
+* Documentation: Clarified documentation about cloning of input arguments in several places.
+* Performance enhancements for Graph concatenation.
+* More informative error outputs.
+* New attribute graph_model `GraphLearner` class, which gets the trained graph.
+* `as_learner()` S3-method for PipeOp class that turns wraps a PipeOp in a Graph and turns that into a Learner.
+* Changed PipeOps:
+  - `PipeOpHistBin`: renamed 'bins' Param to 'breaks'
+  - `PipeOpImputeHist`: fix handling of integer features spanning the entire represented integer range
+  - `PipeOpImputeOOR`: fix handling of integer features spanning the entire represented integer range
+  - `PipeOpProxy`: Avoid unnecessary clone
+  - `PipeOpScale`: Performance improvement
+
```

Comment:
Was ist mit mlr3spatial? https://github.com/mlr-org/mlr3spatial/releases/tag/v0.1.0

Reply:
und mlr3proba https://github.com/mlr-org/mlr3proba/releases/tag/v0.4.2

Reply:
und mlr3 tuning spaces https://github.com/mlr-org/mlr3tuningspaces

Reply:
mlr3tuningspaces wurde 2022 released.

Bei den anderen müsste man im code nachschauen warum diese nicht in die Liste kamen.

Reply:
Magst du mal schauen? Instructions [hier im Wiki](https://github.com/mlr-org/mlr3/wiki/Collecting-quarterly-package-updates)

Reply:
> mlr3tuningspaces wurde 2022 released.

Ne da habe ich nur zu spät den Release auf gh erstellt. Ich füge es mal hinzu

Reply:
Ich guck mir das Skript an

Reply:
Ach du guckst über CRAN nach dem Release Date. Ja mlr3tuningspaces 0.0.1 war im September. Vielleicht lassen wir es aber trotzdem drin. War der inital release und gibt ja keinen q3 Post.

Reply:
Vermutlich erkärt das dann auch mlr3proba und so (ich habe nur auf die GH releases gekuckt)

Reply:
@pat-s gh only returns 30 repos by default. You can use `https://api.github.com/orgs/mlr-org/repos?per_page=100` to get 100 repos. We have 61.

Reply:
Danke, updated!